src/Server/AwsCloudFormationTemplates.fs

```diff
@@ -12,481 +9,221 @@ open Shared
-        logicalId = logicalId
-        importId = importId
-    }
+let requirePart (key: string) (data: Dictionary<string, string>) =
+    if data.ContainsKey key 
+    then Ok data[key]
+    else Error $"Missing required part '{key}'"
 
-let remapTransferServer
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (resourceContext: AwsResourceContext)
-    (spec: CustomRemapSpecification) 
-    : RemappedSpecResult =
-    let data = resourceData[resource.logicalId]
-    let serverPhysicalIdParts = data["Id"].Split("/")
-    let serverId = serverPhysicalIdParts[1]
-    let importId = serverId
-    let resourceType = spec.pulumiType
-    let logicalId = resource.logicalId.Replace("-", "_")
-    {
-        resourceType = resourceType
-        logicalId = logicalId
-        importId = importId
-    }
+let optionalPart (key: string) (data: Dictionary<string, string>) =
+    if data.ContainsKey key 
+    then Some data[key]
+    else None
 
-let remapSpecifications = Map.ofList [
-    "AWS::ApiGateway::Resource" => {
-        pulumiType = "aws:apigateway/resource:Resource"
-        importIdentityParts = ["RestApiId"; "Id"]
-        delimiter = "/"
-        remapFunc = remapFromImportIdentityParts
-        validatorFunc = validateFromImportIdentityParts
-    }
-
-    "AWS::ApiGateway::Stage" => {
-        pulumiType = "aws:apigateway/stage:Stage"
+let importIdentityBuilders : Map<string, ImportIdentityBuilder> = Map.ofList [
+    
+    "AWS::ApiGateway::Resource" => defaultImportIdentity {
         importIdentityParts = ["RestApiId"; "Id"]
         delimiter = "/"
-        remapFunc = remapFromImportIdentityParts
-        validatorFunc = validateFromImportIdentityParts
     }
 
-    "AWS::ApiGateway::Deployment" => {
-        pulumiType = getPulumiType "AWS::ApiGateway::Deployment"
+    "AWS::ApiGateway::Deployment" => defaultImportIdentity {
         importIdentityParts = ["RestApiId"; "Id"]
         delimiter = "/"
-        remapFunc = remapFromImportIdentityParts
-        validatorFunc = validateFromImportIdentityParts
     }
 
-    "AWS::ApiGateway::UsagePlanKey" => {
-        pulumiType = getPulumiType "AWS::ApiGateway::UsagePlanKey"
+    "AWS::ApiGateway::UsagePlanKey" => defaultImportIdentity {
         importIdentityParts = ["UsagePlanId"; "KeyId"]
         delimiter = "/"
-        remapFunc = remapFromImportIdentityParts
-        validatorFunc = validateFromImportIdentityParts
     }
+    
+    "AWS::ApplicationAutoScaling::ScalingPolicy" => fun resource data context ->
+        result {
+            let! scalingTargetId = requirePart "ScalingTargetId" data
+            let! policyName = requirePart "PolicyName" data
+            let! importId = 
+                match scalingTargetId.Split("|") with
+                | scalingTargetParts when scalingTargetParts.Length = 3 -> 
+                    scalingTargetParts
+                    |> Array.rev
```

Comment:
I don't think Array.rev is quite right here - the format should be `[scalingTargetIdParts[2]; scalingTargetIdParts[0]; scalingTargetIdParts[1];` - rev would produce [2][1][0] instead right?

Reply:
Nice catch! I didn't read it right xD